packages/svelte-ux/src/lib/utils/number.test.ts

```diff
@@ -47,3 +47,165 @@ describe('round()', () => {
     expect(actual).equal(123.46);
   });
 });
+
+describe('formatNumber()', () => {
+  it('returns empty string for null', () => {
+    const actual = formatNumber(null);
+    expect(actual).equal('');
+  });
+
+  it('returns empty string for undefined', () => {
+    const actual = formatNumber(undefined);
+    expect(actual).equal('');
+  });
+
+  it('returns value as string for style "none"', () => {
+    const actual = formatNumber(1234.5678, { style: 'none' });
+    expect(actual).equal('1234.5678');
+  });
+
+  it('formats number with integer', () => {
+    const actual = formatNumber(1234.5678, { style: 'integer' });
+    expect(actual).equal('1234');
+    expect(actual).equal('1,234'); // TODO: Today its like this... It's intended? (I leave the test breaking for now)
```

Comment:
It is intended.  I find it more readable to have integer/counts with commas... such a... `12,345 results` vs `12345 results` (or even bigger numbers).

Maybe there is a use case to not have this?  If so, this might be an example where we would want to be able to customize the `style` (presets) via context like we chatted about today.

Reply:
What if we formatted integers using...

```js
Intl.NumberFormat('en-US', {
  maximumSignificantDigits: value?.toString().split('.')[0].length
})
```

see [REPL](https://svelte.dev/repl/bc26d92b40e942b79395889759f92fa1?version=4.2.4)

Reply:
- It's a US thing to have `,` for 1_234 (in europe this would be very confusing)
- the name "integer" almost implies an integer to me (no formats (,) & no rounding)

Actually, what is a `integer` today is: `formatNumber(1234.5678, { fractionDigits: 0 });` _(knowing that default style is decimal)_

I understand the need, just I'm not sure about the terme "integer"